### DISCUSSION TITLE
WireGuard tunnels not connecting over mobile cellular network Host and Client doesn't receive anything

### DISCUSSION THREAD

**Discussion by null-p01ntr** [pivpn/pivpn#1822] [Support] [answered]
I extensively use my mobile phone connected to my home network on the go and as a hotspot for my laptop. Recently I was unable to connect to my VPN through any of my configurations when I use my phone network. I tried with a different sim card too (same carrier) no luck. How should I trace my problem? Is there a known workaround to this?
I find "only connecting with mobile and no connection on wifi" but I'm having the opposite one. I can connect over public and private networks but not with my own mobile network.

More details about my problem (debug output etc.): [https://github.com/pivpn/pivpn/issues/1821](url)
(I live in Turkey, using Vodafone)

**Comment by orazioedoardo** [comment]
> only connecting with mobile and no connection on wifi

You CAN use wifi or whatever connection you want, FAQ however advised to use different network to test rather than the same network where PiVPN is located.

> I can connect over public and private networks but not with my own mobile network.

Try this https://docs.pivpn.io/faq/#tweaking-mtu

**Reply by Janks14** [reply]
Hi could you please assist me as well

**Reply by Janks14** [reply]
So I found a tutorial on how to setup a VPN server using pivpn and a raspberry pi

The setup works . When I'm connected to the internal network I can ping the servers IP address but when I connect from an external network and connect the wireguard VPN it does not ping the pivpn server 

I noticed when I connected the laptop using wireguard on the wireguard app it showed the listen port different to the one I used on the router , plus my router has no port forwarding settings but a virtual server settings? Is that the same thing or is that something else

**Comment by null-p01ntr** [comment]
I don't have a problem using wifi connections (ex: friends and cafe wifis not my home wifi with pivpn)

I found out my local advised MTUs are **1458, 1492, 1467, 1450, 1430.** I change on the mobile app directly and unfortunately **none of them worked**.

How can I find the absolute right value and how can I trace if I'm in the right direction while incrementing and decrementing MTU?
I always get a **handshake fail after 5 seconds. No packets on anyside** I am not experienced in these kinds of networking adjustments.

I tried my sim on different devices too, no luck.

I had the same setup working without MTU tweak for months without any issue.

**Comment by null-p01ntr** [comment]
Can this issue be about port forwarding?

This is my current settings on modem interface. (Always been like this) 
192.168.1.13 is the raspberrypi's IP
![image](https://github.com/pivpn/pivpn/assets/38628377/e95b3078-61ef-4dd7-9bb6-f96236baf4e3)

**Comment by orazioedoardo** [comment]
Can you confirm you can connect from networks other than your cellular network?

**Reply by null-p01ntr** [reply]
I checked with other carriers and wifi networks. I can confirm that connection is successful on Wi-Fi networks and failed on mobile networks. (whether using on mobile or with hotspot)

I also tried changing my APN between v4, v6 and v4/v6 but no luck.

**Reply by orazioedoardo** [reply]
If it works on some network and not on others even after reducing the MTU i'd say those networks may be blocking wireguard.

**Comment by null-p01ntr** [comment, marked as answer]
It seems like it was my carrier blocking the WireGuard protocol.
I found an expert on the topic who had the same issue as me with the same carrier. He had a solution by bypassing the protocol censoring.
His instructions: [https://keremerkan.net/posts/udp2raw-bypass-censoring-wireguard-protocol/](url).
I didn't try it yet because it doesn't fit my use case, I will try OpenVPN for now and see how it goes

**Result** 
Reinstalling PiVPN with OpenVPN worked for me.
What a shame to be left out of modern technology because of biased policies.

**Reply by bugrakaann** [reply]
I am also living in Turkey and using Vodafone. I have the same problem. Have you found any solution?

**Reply by null-p01ntr** [reply]
@bugrakaann ;
I switched to OpenVPN (reinstalled pivpn and choose ovpn for protocol), it's working fine since then. The solution I referenced seemed too advanced for my needs and didn't cover some of my needs.

> It seems like it was my carrier blocking the WireGuard protocol. I found an expert on the topic who had the same issue as me with the same carrier. He had a solution by bypassing the protocol censoring. His instructions: [https://keremerkan.net/posts/udp2raw-bypass-censoring-wireguard-protocol/](url). I didn't try it yet because it doesn't fit my use case, I will try OpenVPN for now and see how it goes
